packages/sdk/src/protocols/base_protocol.ts

```diff
@@ -32,6 +39,19 @@ export class BaseProtocolSDK implements IBaseProtocolSDK {
     options: Options
   ) {
     this.log = new Logger(`sdk:${core.multicodec}`);
+
+    if (core.multicodec.includes("filter")) {
```

Comment:
I think this is needed only for `healthManager` so I would advice to do:
```ts
const healthManager = HealthManager.getInstance(core.multicodec);
...
healthManager.update(this.peers.length);
```

because `this.name` is not used anywhere it should be incapsulated to `healthManager`

Reply:
Agree that `name` can be useful, but if it is not used - let's make it internal for now.